activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -240,14 +240,31 @@ def truncate_all(environment = env)
         end
       end
 
+      def migrate_all
+        db_configs = ActiveRecord::Base.configurations.configs_for(env_name: ActiveRecord::Tasks::DatabaseTasks.env)
+        db_configs.each { |db_config| initialize_database(db_config) }
+
+        if db_configs.size == 1 && db_configs.first.primary?
+          ActiveRecord::Tasks::DatabaseTasks.migrate
+        else
+          mapped_versions = ActiveRecord::Tasks::DatabaseTasks.db_configs_with_versions
+
+          mapped_versions.sort.each do |version, db_configs|
+            db_configs.each do |db_config|
+              ActiveRecord::Tasks::DatabaseTasks.with_temporary_connection(db_config) do
+                ActiveRecord::Tasks::DatabaseTasks.migrate(version)
+              end
+            end
+          end
+        end
+      end
+
       def migrate(version = nil)
         scope = ENV["SCOPE"]
         verbose_was, Migration.verbose = Migration.verbose, verbose?
 
         check_target_version
 
-        initialize_database(migration_connection_pool.db_config)
```

Comment:
It's possible because this is a public API that someone is depending on this behavior here, maybe we can pass in a kwarg like `skip_initialize`?

Reply:
:+1: `skip_initialize` added - thx for the review.